GiniVision/Classes/GINICamera.swift

```diff
@@ -16,80 +16,80 @@ internal class GINICamera {
+            if device.isFocusPointOfInterestSupported && device.isFocusModeSupported(focusMode) {
                 device.focusPointOfInterest = point
                 device.focusMode = focusMode
             }
             
-            if device.exposurePointOfInterestSupported && device.isExposureModeSupported(exposureMode) {
+            if device.isExposurePointOfInterestSupported && device.isExposureModeSupported(exposureMode) {
                 device.exposurePointOfInterest = point
                 device.exposureMode = exposureMode
             }
             
-            device.subjectAreaChangeMonitoringEnabled = monitorSubjectAreaChange
+            device.isSubjectAreaChangeMonitoringEnabled = monitorSubjectAreaChange
             device.unlockForConfiguration()
         })
     }
     
-    func captureStillImage(completion: (inner: () throws -> NSData) -> ()) {
-        dispatch_async(sessionQueue, {
+    func captureStillImage(_ completion: @escaping (_ inner: () throws -> Data) -> ()) {
+        sessionQueue.async(execute: {
             // Connection will be `nil` when there is no valid input device; for example on iOS simulator
-            guard let connection = self.stillImageOutput?.connectionWithMediaType(AVMediaTypeVideo) else {
-                return completion(inner: { _ in throw GINICameraError.NoInputDevice })
+            guard let connection = self.stillImageOutput?.connection(withMediaType: AVMediaTypeVideo) else {
+                return completion({ _ in throw GINICameraError.noInputDevice })
             }
             // Set the orientation accoding to the current orientation of the device
             if let orientation = AVCaptureVideoOrientation(self.motionManager.currentOrientation) {
                 connection.videoOrientation = orientation
             } else {
-                connection.videoOrientation = .Portrait
+                connection.videoOrientation = .portrait
             }
-            self.videoDeviceInput?.device.setFlashModeSecurely(.On)
-            self.stillImageOutput?.captureStillImageAsynchronouslyFromConnection(connection, completionHandler: { (imageDataSampleBuffer: CMSampleBuffer!, error: NSError!) -> Void in
-                guard error == nil else { return completion(inner: { _ in throw GINICameraError.CaptureFailed }) }
+            self.videoDeviceInput?.device.setFlashModeSecurely(.on)
+            self.stillImageOutput?.captureStillImageAsynchronously(from: connection, completionHandler: { (imageDataSampleBuffer: CMSampleBuffer?, error: Error?) -> Void in
+                guard error == nil else { return completion({ _ in throw GINICameraError.captureFailed }) }
                 let imageData = AVCaptureStillImageOutput.jpegStillImageNSDataRepresentation(imageDataSampleBuffer)
-                completion(inner: { _ in return imageData })
+                completion({ _ in return imageData! })
```

Comment:
Not sure what kind of problems could arise here, but I'm not a fan of force unwrapping here.